api.go

```diff
@@ -190,17 +196,74 @@ func BootstrapCluster(config *Config, logs LogStore, stable StableStore, snaps S
 		return fmt.Errorf("Bootstrap only works on new clusters, found snapshots")
 	}
 
-	// Set current term to 1
+	// Set current term to 1.
 	if err := stable.SetUint64(keyCurrentTerm, 1); err != nil {
 		return fmt.Errorf("failed to save current term: %v", err)
 	}
 
-	// Append configuration entry to log
+	// Append configuration entry to log.
 	entry := &Log{
 		Index: 1,
 		Term:  1,
-		Type:  LogConfiguration,
-		Data:  encodeConfiguration(configuration),
+	}
+	if conf.ProtocolVersion < 1 {
```

Comment:
During the phase where we are transitioning to protocol version 1 but have set it to 0, we may not want to have this new entry in the log in case an older server joins, since we are saying if protocol version == 0 that we are compatible with old servers.